### PR TITLE
Harmonize system update for SLE and openSUSE

### PR DESCRIPTION
**Pull request by rwx788**
See [poo#31954](https://progress.opensuse.org/issues/31954).

#### Verification runs
 * [sle 15 textmode](http://g226.suse.de/tests/2612#step/zypper_up/1)
 * [sle 15 gnome](http://g226.suse.de/tests/2626)
 * [sle 15 staging](http://g226.suse.de/tests/2636#)
 * [sle 12 SP4 gnome](http://g226.suse.de/tests/2629#step/updates_packagekit_gpk/19)
 * [openSUSE staging textmode](http://g226.suse.de/tests/2612#step/zypper_up/1)
 * [openSUSE staging kde](http://g226.suse.de/tests/2637#)
 * [qam all patterns](http://g226.suse.de/tests/2630)
 * [leap 15.o updates kde](http://g226.suse.de/tests/2633)
 * [TW kde](http://g226.suse.de/tests/2631)
 * [qam-regression-other](http://g226.suse.de/tests/2639)
 * [mau-extratests](http://g226.suse.de/tests/2641)

### CODE REVIEW

**Comment by rwx788** [comment]
@andreasstieger @lnussel @DimStar77 @nilxam @mimi1vx @czerw, may I kindly ask you to take a look on this PR, and point me to the jobs which may get broken by it, so I could add another verification run. 

The only expected change in the schedule, is that we execute `update/prepare_system_for_update_tests` on SLE, which we haven't done before. For all other scenarios schedule is identical.

**Comment by mimi1vx** [comment]
for QAM this look harmless, in this phase of tests is system probably in most cases fully updated with released updates + added incidents patches

**Comment by czerw** [comment]
It looks fine from SLE/QAM point of view, as we use different approach to updates(we don't use zypper_up). Anyway, please do verification for https://openqa.suse.de/tests/2092879  and  https://openqa.suse.de/tests/2088713.

File: lib/main_common.pm [hunk 1]
```diff
@@ -2087,8 +2087,17 @@ sub guiupdates_is_applicable {
...
     my (%args) = @_;
     my $console_updates = $args{console_updates} // 0;
-
+    # Do not run if system is updated already
     return if get_var('SYSTEM_UPDATED');
+
+    if (is_sle) {
+        # Do not run on sle staging, as doesn't make sense there
+        return if is_staging;
+        # On SLE we schedule console and x11 tests together
```
[the remark this answers is not in the document]
Comment (rwx788) [reply lib/main_common.pm]: Yes, but in openSUSE we trigger those as a first thing after installation, which is totally different approach, as system gets updated before any test.

File: lib/main_common.pm [hunk 2]
```diff
@@ -2100,27 +2109,24 @@ sub load_system_update_tests {
...
+                loadtest "update/updates_packagekit_gpk";
+                set_var('SYSTEM_UPDATED', 1);
             }
-            elsif (x11tests_is_applicable()) {
-                unless (is_sle && is_staging) {
-                    loadtest "update/updates_packagekit_gpk";
-                    set_var('SYSTEM_UPDATED', 1);
-                }
-            }
+            # Test was never executed on SLE and fails currently
```
[the remark this answers is not in the document]
Comment (rwx788) [reply lib/main_common.pm]: I tried to make it work, problem is that it's console test, which is executed either between x11 tests or console tests. As this is schedule used in most of the places, I would not change it.

**Comment by rwx788** [comment]
@czerw thanks for the feedback

File: tests/update/prepare_system_for_update_tests.pm [hunk 3]
```diff
@@ -20,13 +20,15 @@ use base "consoletest";
...
 use testapi;
 use utils;
+use version_utils 'is_sle';
 
 sub run {
     select_console 'root-console';
     ensure_serialdev_permissions;
-
-    assert_script_run "echo \"download.use_deltarpm = false\" >> /etc/zypp/zypp.conf";
+    # default is true, for legacy reasons we were running this on openSUSE only
```
[the remark this answers is not in the document]
Comment (rwx788) [reply tests/update/prepare_system_for_update_tests.pm]: No, I have no idea. And again, based on my experience, I'm reluctant to introduce such changes. 
So I kept old behavior.

**Comment by asmorodskyi** [comment]
LGTM

**Pull request merged**